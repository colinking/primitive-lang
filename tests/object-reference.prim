// SYNTAX TEST "source.prim" "object with type references"

namespace airplane

object TestObject123 {
  field1 airplane.TestObject456
}
