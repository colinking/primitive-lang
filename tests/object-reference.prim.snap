>// SYNTAX TEST "source.prim" "object with type references"
#^^ source.prim comment.line.double-slash.prim punctuation.definition.comment.prim
#  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.prim comment.line.double-slash.prim
>
>namespace airplane
#^^^^^^^^^ source.prim keyword.other.namespace.prim
#         ^ source.prim
#          ^^^^^^^^ source.prim entity.name.namespace.prim
>
>object TestObject123 {
#^^^^^^ source.prim keyword.other.object.prim
#      ^ source.prim meta.object.prim
#       ^^^^^^^^^^^^^ source.prim meta.object.prim entity.name.type.prim
#                    ^ source.prim meta.object.prim
#                     ^ source.prim meta.object.prim punctuation.definition.block.prim
>  field1 airplane.TestObject456
#^^ source.prim meta.object.prim meta.field.declaration.prim
#  ^^^^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.prim
#        ^ source.prim meta.object.prim meta.field.declaration.prim
#         ^^^^^^^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.type.prim entity.name.namespace.prim
#                  ^^^^^^^^^^^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.type.prim
>}
#^^ source.prim meta.object.prim meta.field.declaration.prim
>